src/picosvg/svg.py

```diff
@@ -891,6 +889,99 @@ def normalize_opacity(self, inplace=False):
 
         return self
 
+    def _iter_nested_svgs(
+        self, root: etree.Element
+    ) -> Generator[etree.Element, None, None]:
+        # This is different from Element.iter("svg") in that we don't yield the root
+        # svg element itself, only traverse its children and yield any immediate
+        # nested SVGs without traversing the latter's children as well.
+        frontier = deque(root)
+        while frontier:
+            el = frontier.popleft()
+            if el.tag is etree.Comment:
+                continue
+            if strip_ns(el.tag) == "svg":
+                yield el
+            elif len(el) != 0:
+                frontier.extend(el)
+
+    def _unnest_svg(
+        self, svg: etree.Element, parent_width: float, parent_height: float
+    ) -> Tuple[etree.Element, ...]:
+        x = float(svg.attrib.get("x", 0))
+        y = float(svg.attrib.get("y", 0))
+        width = float(svg.attrib.get("width", parent_width))
+        height = float(svg.attrib.get("height", parent_height))
+
+        viewport = viewbox = Rect(x, y, width, height)
+        if "viewBox" in svg.attrib:
+            viewbox = parse_view_box(svg.attrib["viewBox"])
+
+        # first recurse to un-nest any nested nested SVGs
+        self._swap_elements(
+            (el, self._unnest_svg(el, viewbox.w, viewbox.h))
+            for el in self._iter_nested_svgs(svg)
+        )
+
+        g = etree.Element(f"{{{svgns()}}}g")
+        for el in svg:
+            g.append(el)
+
+        if viewport != viewbox:
+            preserve_aspect_ratio = svg.attrib.get("preserveAspectRatio", "xMidYMid")
+            transform = Affine2D.rect_to_rect(viewbox, viewport, preserve_aspect_ratio)
+        else:
+            transform = Affine2D.identity().translate(x, y)
+
+        if "transform" in svg.attrib:
+            transform = Affine2D.compose_ltr(
+                (transform, Affine2D.fromstring(svg.attrib["transform"]))
+            )
+
+        if transform != Affine2D.identity():
+            g.attrib["transform"] = transform.tostring()
+
+        # TODO Define a viewport-sized clipPath once transform+clip-path issue is fixed
+        # https://github.com/googlefonts/picosvg/issues/200
+        return (g,)
+
+    def resolve_nested_svgs(self, inplace=False):
+        """Replace nested <svg> elements with equivalent <g> with a transform.
+
+        NOTE: currently this is still missing two features:
```

Comment:
Should clarify what it does if it encounters the unsupported things: not-quite-right answer or error?

Reply:
no error but not-quite-right-answer (we don't clip and don't resolve percents relative to the correct ancestor)